src/ja/server/dispatcher/Dispatcher.py

```diff
@@ -0,0 +1,36 @@
+from ja.common.job import Job
+
+
+class Dispatcher:
+    """
+    Represents the mediator between the Scheduler and the work machines.
+    """
+
+    def __init___(self):
+        """!
+        Constructor for the dispatcher class
+        """
+
+    def set_distribution(self, jobs: list[Job]) -> None:
+        """!
+        Distributes the newly added jobs to the work machines
+        and potentially pauses jobs that need to be paused.
+
+        @param jobs list of the new job distribution.
+        """
+
+    def _get_changed_jobs(self) -> list[Job]:
```

Comment:
I suppose you'd need to pass the new distribution of jobs as a parameter so that this function can compare the old and the new distributions?